src/differentials.jl

```diff
@@ -227,6 +228,33 @@ mul_dne(::DNE, ::DNE) = DNE()
 mul_dne(::DNE, ::Any) = DNE()
 mul_dne(::Any, ::DNE) = DNE()
 
+#####
+##### `NotImplemented`
+#####
+
+"""
+    NotImplemented <: AbstractDifferential
+
+A differential type which behaves similar to [`DNE`](@ref) but instead signifies that
+the actual differential is not implemented in ChainRules, not that it does not exist.
```

Comment:
Can there be a "not yet" added here?
Maybe even a "PRs to remove not implemented rules are welcome"

Reply:
Good idea.